### ISSUE TITLE
Weird css in some rows of ACA search results

### ISSUE THREAD

**Issue by sergiobaiao** [Alfresco/alfresco-content-app#4254]
**Describe the bug**
I'm having some weird ACA search results within my environment, like the picture bellow:

![image](https://github.com/user-attachments/assets/ddfe5473-91bb-4990-b518-70f0714c1952)

Some rows in the results are showing "n class='aca-highlight'>" within the title, description and/or content. Aparently this only happens to my own Alfresco data, since @aborroy was able to replicate this with his own local data, but did replicate with my data from my Alfresco server.

**Environment**
- ACA version: 5.2.0
- Alfresco 23.2 from alfresco-docker-installer with this .env

ALFRESCO_CE_TAG=23.2.1
SEARCH_CE_TAG=2.0.9.1
SHARE_TAG=23.2.1
ACA_TAG=4.4.1
POSTGRES_TAG=15.6
#POSTGRES_TAG=14.4
MARIADB_TAG=11.3.2
TRANSFORM_ENGINE_TAG=5.1.0
ACTIVEMQ_TAG=5.18-jre17-rockylinux8

**To Reproduce**
1. Go to ACA
2. Click on search icon and perform a search
3. Scroll down to search results

I don't think this will happen to everyone, maybe it's related to wrong character encoding from metadata on the search results, that maybe wrongly escaping quotes or any other chars. Also, the last line from each row is showing a lot of weird chars also.

**Screenshots**
ACA Search Results:
![image](https://github.com/user-attachments/assets/aac99765-8916-4959-9525-14eaf08653ae)

Share Search Results: (same user and same search term)
![image](https://github.com/user-attachments/assets/892e47af-32c8-48c3-a18c-0556c42b40a5)

P.S - I can provide a Consumer only user with restricted access to anyone who wants to test this against my Alfresco server.

**Comment by aborroy** [comment]
@DenysVuika my guessing is that there is some error in this method:

https://github.com/Alfresco/alfresco-content-app/blob/5.2.0/projects/aca-content/src/lib/components/search/search-results-row/search-results-row.component.ts#L157C1-L161C4

Some inputs, like a nested "aca-highlight" input would create this issue.

A better approach could be to use HTML elements for this replacement. Something similar to:

`// Using a proper HTML parser
private stripHighlighting(highlightedContent: string): string {
  if (!highlightedContent) return '';
  
  const parser = new DOMParser();
  const doc = parser.parseFromString(highlightedContent, 'text/html');
  const highlights = doc.querySelectorAll('.aca-highlight');
  
  highlights.forEach(highlight => {
    const parent = highlight.parentNode;
    if (parent) {
      while (highlight.firstChild) {
        parent.insertBefore(highlight.firstChild, highlight);
      }
      parent.removeChild(highlight);
    }
  });
  
  return doc.body.innerHTML;
}`

**Comment by sergiobaiao** [comment]
while researching today for a probable cause, i've done this to log what i was receiving from the search results:
   const highlights: SearchEntryHighlight[] = this.node.entry['search']?.['highlight'];
    let name = this.node.entry.name;
    console.log("name", name);
    const properties = this.node.entry.properties;
    console.log("properties", properties);
    let title = properties?.['cm:title'] || '';
    console.log("title", title);
    let description = properties?.['cm:description'] || '';
    console.log("description", description);
    let content = '';
    console.log("content", content);


and this is the search result:
![image](https://github.com/user-attachments/assets/ddb44447-4f18-4bf3-9c46-1b76be74e348)


If you look what i've circled in red in the right side, the description field has a lot of back/forward slashes, and this entire piece is being stripped from the search result on the left site. I'm not sure if the slashes are being correctly escaped and this may be one of the causes.

**Comment by sergiobaiao** [comment]
> @DenysVuika my guessing is that there is some error in this method:
> 
> https://github.com/Alfresco/alfresco-content-app/blob/5.2.0/projects/aca-content/src/lib/components/search/search-results-row/search-results-row.component.ts#L157C1-L161C4
> 
> Some inputs, like a nested "aca-highlight" input would create this issue.
> 
> A better approach could be to use HTML elements for this replacement. Something similar to:
> 
> ```
> private stripHighlighting(highlightedContent: string): string {
>   if (!highlightedContent) return '';
>   
>   const parser = new DOMParser();
>   const doc = parser.parseFromString(highlightedContent, 'text/html');
>   const highlights = doc.querySelectorAll('.aca-highlight');
>   
>   highlights.forEach(highlight => {
>     const parent = highlight.parentNode;
>     if (parent) {
>       while (highlight.firstChild) {
>         parent.insertBefore(highlight.firstChild, highlight);
>       }
>       parent.removeChild(highlight);
>     }
>   });
>   
>   return doc.body.innerHTML;
> }
> ```

I tried this and it still shows the same error.

**Comment by aborroy** [comment]
In that case, the fix should be something like:

```
private stripHighlighting(highlightedContent: string): string {
  if (!highlightedContent) return '';
  
  // Escape special regex characters in prefix and postfix
  const escapedPrefix = this.highlightPrefix.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
  const escapedPostfix = this.highlightPostfix.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
  
  return highlightedContent
    .replace(new RegExp(escapedPrefix, 'g'), '')
    .replace(new RegExp(escapedPostfix, 'g'), '');
}
```

**Comment by sergiobaiao** [comment]
> In that case, the fix should be something like:
> 
> ```
> private stripHighlighting(highlightedContent: string): string {
>   if (!highlightedContent) return '';
>   
>   // Escape special regex characters in prefix and postfix
>   const escapedPrefix = this.highlightPrefix.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
>   const escapedPostfix = this.highlightPostfix.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
>   
>   return highlightedContent
>     .replace(new RegExp(escapedPrefix, 'g'), '')
>     .replace(new RegExp(escapedPostfix, 'g'), '');
> }
> ```

tried that also, didn't work.

**Comment by sergiobaiao** [comment]
any workaround on this?

**Comment by sergiobaiao** [comment]
i can do better...is there a way i can contact you directly so i can send you an username/password to my alfresco server?

**Comment by MichalKinas** [comment]
Yes you can message me on MS Teams, Slack or send me an email -> Michal.Kinas@hyland.com

**Comment by sergiobaiao** [comment]
Hi Michal, did you receive my email?

**Comment by MichalKinas** [comment]
Okay I analyzed what's going on there and it seems that this is an issue with search engine, instead of providing proper prefix `<span class='aca-highlight'>` for every highlight it seems to apply `n class='aca-highlight'>` instead which break the rendering in ACA in that particular case, I will forward this issue to the BE team to analyze it

**Comment by MichalKinas** [comment]
@sergiobaiao could I use one of the PDFs for which the issue is observed and attach it to the bug report for the BE team? So that they can reproduce it locally in their ACS instance? Also please let me know which search engine you're using in your instance, is it SOLR or ES?

**Comment by sergiobaiao** [comment]
Yes you can, these are public domain documents. We're using SOLR.

**Comment by MichalKinas** [comment]
Okay so I was able to reproduce this issue locally with Solr, I created the bug for the BE team and passed the info for them

**Comment by MichalKinas** [comment]
I'm sorry but we cannot grant you access there, this bug will be picked up by the BE team as soon as this will be possible, I will update you on the progress and planned release date here

**Comment by sergiobaiao** [comment]
Hi there Michal, are there any updates on this issue?

**Comment by MichalKinas** [comment]
Hello Sergio, as far as I can see the BE team already picked this up so I hope it will be fixed until next release of ACA

**Comment by sergiobaiao** [comment]
Good morning Michal, any news? I've tried the latest ACA (6.0.0) but still shows the error. I think i need the complete update system (ACS and SOLR), is that right?

**Comment by MichalKinas** [comment]
Hello Sergio, we've just finished the release process, yes to get the fix you need to use ACS and SOLR from latest release - 25.1 for ACS

**Comment by MichalKinas** [comment]
@sergiobaiao could you please confirm that the issue is resolved now so that I can close the issue?

**Comment by MichalKinas** [comment]
@sergiobaiao Thanks for the response, in that case I think this is no longer an ACA issue and FE team won't be able to fix that, I think that the best way to proceed would be to close this issue in ACA and report this latest problem in https://github.com/Alfresco/SearchServices so that our BE team can investigate that, does that work for you?
